crates/ra_hir/src/expr.rs

```diff
@@ -103,6 +107,18 @@ impl BodySyntaxMapping {
     }
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum Literal {
+    String(String),
+    ByteString(Vec<u8>),
+    Char(char),
+    Bool(bool),
+    Int(u64, UncertainIntTy),
+    Float(u64, UncertainFloatTy), // FIXME: f64 is not Eq
+    Tuple { values: Vec<ExprId> },
+    Array { values: Vec<ExprId> },
```

Comment:
I don't know if I put count Tuple and Array with the other literals, since they have subexpressions... seems simpler to me to make them separate `Expr` variants